components/support/sync15-traits/src/client.rs

```diff
@@ -23,38 +24,88 @@ pub struct RemoteClient {
     pub device_type: Option<DeviceType>,
 }
 
-/// The type of a client. Please keep these variants in sync with the device
-/// types in the FxA client, tabs and sync manager.
-// In particular, ensure the to and from string values (either explicitly as
-// below, or via serde as done by fxa-client) match.
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+/// Enumeration for the different types of device.
+///
+/// Firefox Accounts and the broader Sync universe separates devices into broad categories for
+/// display purposes, such as distinguishing a desktop PC from a mobile phone.
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub enum DeviceType {
+    #[serde(rename = "desktop")]
     Desktop,
+    #[serde(rename = "mobile")]
     Mobile,
+    #[serde(rename = "tablet")]
     Tablet,
+    #[serde(rename = "vr")]
     VR,
+    #[serde(rename = "tv")]
     TV,
+    #[serde(other)]
+    #[serde(skip_serializing)] // Don't you dare trying.
+    Unknown,
 }
 
 impl DeviceType {
-    pub fn try_from_str(d: impl AsRef<str>) -> Option<DeviceType> {
-        match d.as_ref() {
-            "desktop" => Some(DeviceType::Desktop),
-            "mobile" => Some(DeviceType::Mobile),
-            "tablet" => Some(DeviceType::Tablet),
-            "vr" => Some(DeviceType::VR),
-            "tv" => Some(DeviceType::TV),
-            _ => None,
+    // Tried to `impl From<DeviceType> for Option<DeviceType>` but that conflicts with
+    // `impl<T> From<T> for Option<T>` in the core.
+    pub fn into(dt: DeviceType) -> Option<DeviceType> {
+        // Turn DeviceType::Unknown into None, everything else into Some()
+        match dt {
+            DeviceType::Unknown => None,
+            _ => Some(dt),
```

Comment:
Having both `Some(DeviceType::Unknown)` and `None` as possibilities seems like a potential footgun, but I'm guessing it's needed to support some legacy code?  Could you add a note here explaining the situation?

Reply:
good call! It's actually *this* which is legacy code - while I just moved this from somewhere, this isn't used, so I killed it.